src/modelmanager.cpp

```diff
@@ -221,6 +222,47 @@ Status ModelManager::loadPipelinesConfig(rapidjson::Document& configJson) {
     return ovms::StatusCode::OK;
 }
 
+Status ModelManager::loadCustomLoadersConfig(rapidjson::Document& configJson) {
+    const auto itrp = configJson.FindMember("custom_loader_config_list");
+    if (itrp == configJson.MemberEnd() || !itrp->value.IsArray()) {
+        SPDLOG_ERROR("Configuration file doesn't have custom loader property.");
+        return StatusCode::OK;
+    }
+
+    for (const auto& configs : itrp->value.GetArray()) {
+        // Load Customer Loaders as per the configuration
+        const std::string loaderName = configs["config"]["loader_name"].GetString();
+        SPDLOG_INFO("Reading Custom Loader:{} configuration", loaderName);
+
+        CustomLoaderConfig loaderConfig;
+        auto status = loaderConfig.parseNode(configs["config"]);
+        if (status != StatusCode::OK) {
+            SPDLOG_ERROR("Parsing loader:{} config failed", loaderName);
+            return StatusCode::UNKNOWN_ERROR;
+        }
+
+        // this is where library or custom loader is loaded
+        void* handleCL = dlopen(const_cast<char*>(loaderConfig.getLibraryPath().c_str()), RTLD_LAZY | RTLD_GLOBAL);
+        if (!handleCL) {
+            SPDLOG_ERROR("Cannot open library:  {} {}", loaderConfig.getLibraryPath(), dlerror());
+            return StatusCode::UNKNOWN_ERROR;
+        }
+
+        // load the symbols
+        create_t* customObj = (create_t*)dlsym(handleCL, "create");
+        const char* dlsym_error = dlerror();
+        if (dlsym_error) {
+            SPDLOG_ERROR("Cannot load symbol create:  {} ", dlsym_error);
+            return StatusCode::UNKNOWN_ERROR;
+        }
+
+        std::shared_ptr<CustomLoaderInterface> customerLoaderIfPtr{customObj()};
+        customerLoaderIfPtr->loaderInit(const_cast<char*>(loaderConfig.getLoaderConfigFile().c_str()));
+        customLoaderInterfacePtrs.insert({loaderName, customerLoaderIfPtr});
```

Comment:
What if there already is loader with such name? We currently monitor changes for config.json file and re-run `loadConfig` therefore we need to take care of custom loaders reloading.

Reply:
There is another comment against loader re-loading. Having custom loader reloading the library will open up quite a few scenarios like existing black list, state of loaded models etc. We can restrict changing the "customloader" definition section in config file. If the loader with same name already exists at config reload, the section will be skipped. If we need to implement the unloading and reloading we need to come up with all possible cases,